src/metrics/kms-service.ts

```diff
@@ -1,36 +1,21 @@
-// This only imports the type of KMS, not the class. As we don't do an instantiation typescript will compile this away.
-import { KMS } from "aws-sdk";
-
-import { logError } from "../utils";
-
 // In order to avoid the layer adding the 40mb aws-sdk to a deployment, (which is always available
 // in the lambda environment anyway), we use require to import the sdk, and return an error if someone
 // tries to decrypt a value.
 export class KMSService {
-  private kms?: KMS;
 
-  constructor() {
+  public async decrypt(value: string): Promise<string> {
```

Comment:
If somebody installs the extension and uses datadog kms api key variable, decrypt will still get called even though the library doesn't end up using the key. Might be another potential optimization for clients using KMS.

Reply:
absolutely ! maybe in an other PR ?

Reply:
Sure, I don't think it has to be done right now.

Reply:
OK I'll create a ticket for that